_posts/2024-07-09-datafusion-40.0.0.md

```diff
@@ -0,0 +1,450 @@
+benchmarks.
+
+[specialization for single Uft8/LargeUtf8/Binary/LargeBinary]: https://github.com/apache/datafusion/pull/8827
+
+We are also in the final phases of our initial integration of the new [Arrow
+StringView] which will provide a significant performance improvement
+for many workloads. This feature should be available in future versions of DataFusion.
+Kudos to [@XiangpengHong], [@PsiACE], [@Weigju>XXXX] and
+[@AriesDevil], and [@alamb] for driving this along.
+
+[Arrow StringView]: https://docs.rs/arrow/latest/arrow/array/struct.GenericByteViewArray.html
+
+
+# Improved Quality 📋
+
+DataFusion continues to improve overall in quality. One of the most exciting
+improvements is the addition of a new [SQLancer] based [DataFusion Fuzzing]
+suite thanks to [@2010YOUY01] that has already found several bugs (kudos to [@jonahgao],
+YYY, and ZZZ for fixing them so fast).
+
+[SQLancer]: https://github.com/apache/datafusion/issues/11030
+[DataFusion Fuzzing]: https://github.com/datafusion-contrib/datafusion-sqllancer
+[@2010YOUY01]: https://github.com/2010YOUY01
+
+
+## Improved Documentation 📚
+
+We continue to improve the documentation to make it easier to get started using DataFusion with
+the [Library Users Guide], [API documentation], and [Examples].
+
+Some notable new examples include:
+* [sql_analysis.rs] to analyse SQL queries with DataFusion structures (thanks [@LorrensP-2158466])
+* [plan_to_sql.rs] to generate SQL from DataFusion Expr and LogicalPlan (thanks [@edmondop])
+
+[Library Users Guide]: https://datafusion.apache.org/library-user-guide/index.html
+[API documentation]: https://docs.rs/datafusion/latest/datafusion/index.html
+[Examples]: https://github.com/apache/datafusion/tree/main/datafusion-examples
+[sql_analysis.rs]: https://github.com/apache/datafusion/blob/main/datafusion-examples/examples/sql_analysis.rs
+[plan_to_sql.rs]: https://github.com/apache/datafusion/blob/main/datafusion-examples/examples/plan_to_sql.rs
+[@LorrensP-2158466]: https://github.com/LorrensP-2158466
+[@edmondop]: https://github.com/edmondop
+
+# New Features ✨
+
+There are too many new features in the last 6 months to list them all, but here
+are some highlights:
+
+## SQL 
+* Support for unnest (TODO LINK)
+* Support  Recursive CTEs https://github.com/apache/datafusion/pull/9619 / https://github.com/apache/datafusion/issues/462
+* Support for `CREATE FUNCTION` (see below)
+* New functions: TODO find list
+* Improved support for structured types such a  `STRUCT`, `LIST`/`ARRAY` and `MAP` 
+
+```rust
+> select {'foo': {'bar': 2}};
++--------------------------------------------------------------+
+| named_struct(Utf8("foo"),named_struct(Utf8("bar"),Int64(2))) |
++--------------------------------------------------------------+
+| {foo: {bar: 2}}                                              |
++--------------------------------------------------------------+
+1 row(s) fetched.
+Elapsed 0.002 seconds.
+```
+
+## SQL Unparser
```

Comment:
Since the feature/ code refers to this process as Unparsing I think leaving that in is important. 

I have changed the title to  be `## SQL Unparser (SQL Formatter)`